testsuite_overview.md

```diff
@@ -0,0 +1,310 @@
+| delete | x | x | x | | 
+| batch-delete | x | | x | | 
+| action | o |  o | o | x | 
+
+"x" - test is included in test suite  
+"o" - test is not included but method should be supported by the resource  
+" " - test is not included and method should NOT be supported by resource
+
+##### Resource Key Tests
+| Key Feature | Rest.li Method used|
+|-------------|---------------|
+|Key with Union| get | 
+|Query Params| get | 
+|Complex Key| get, create, delete, update batch-create, batch-delete, batch-get, batch-update, partial-update |
+|Special Chars in ComplexKey strings | get, batch-get|
+
+##### Error tests
+| Error | Resource used | Rest.li Method | Details |
+|-----|-----|----|----|
+|404 | Collection | get | Send empty get request|
+|400 | Collection | update | Request has a missing required field | 
+|500 | Collection | create | Create request with id field |
+|Error Details | Collection | create | CreateResponse with Error Details| 
+|Batch Results with Errors | Collection | batch_update | Batch update with one good and two bad requests| 
+
+##### Misc. Tests
+| Feature | Resource tested| Method used|
+|---------|---------|------|
+| Typeref | Collection, Association | get |
+| Subresource | Collection, Association | get |
+| Projection | Collection, Association, ComplexKeyResource | get | 
+
+
+Development Status
+------------------
+* [x] Basic JSON tests added
+* [x] Basic schema tests added
+* [x] Basic collection request/response tests added
+* [x] Basic complex key request/response tests added
+* [x] Basic association request/response tests added
+* [x] Basic simple request/response tests added
+* [x] Basic actionset request/response tests added
+* [x] 404, validation error and other client side error tests added
+* [x] 500 and other server side error tests added
+* [x] Batch tests added
+* [x] Partial update tests added
+* [x] Projection tests added
+* [x] Subresource tests added
+* [x] Unions and maps and lists in complex keys tests added
+* [x] Finder tests added
+* [x] complex action tests added
+* [x] Query params on methods tests added
+* [x] Unions and maps and lists in query params tests added
+* [x] Error details tests added
+* [x] Batch results with errors tests added
+* [x] partial update of a nested field
+* [x] special chars in strings tests added
+* [x] test finder that uses association entity key part
+* [x] test finder with metadata
+* [x] Resource Typeref support tests added
+* [x] generate 2.0.0 wire protocol requests for all tests into separate directories
+* [x] Union of same types in complex types added
+* [x] Basic request/response test for collection with returning entity in create method added
+* [ ] add test for record with lowercase name 
+* [ ] add test for tunneled query params
+* [ ] add test that includes unicode (non-ascii) characters on wire
+* [ ] make input for wire protocol tests language independent
+* [ ] expand manual assertions for Java test suite
+
+
+Troubleshooting: What should I do when...
```

Comment:
I prefer that this should not be in the Overview documentation, may be one for itself.